src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,40 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            String[] parts = line.split(",");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException("Invalid data format: " + line);
+            }
+
+            FruitTransaction.Operation operation = getOperation(parts[0]);
+            String fruit = parts[1];
+            int quantity;
+            try {
+                quantity = Integer.parseInt(parts[2]);
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Invalid quantity: " + parts[2]);
+            }
+
+            FruitTransaction transaction = new FruitTransaction();
+            transaction.setOperation(operation);
+            transaction.setFruit(fruit);
+            transaction.setQuantity(quantity);
+            transactions.add(transaction);
+        }
+        return transactions;
+    }
+
+    private FruitTransaction.Operation getOperation(String code) {
+        for (FruitTransaction.Operation operation : FruitTransaction.Operation.values()) {
+            if (operation.getCode().equals(code)) {
+                return operation;
+            }
+        }
+        throw new IllegalArgumentException("Unknown operation: " + code);
```

Comment:
Throwing 'IllegalArgumentException' here is correct, but it would be better to create a custom exception to provide a more informative message and to differentiate between different types of illegal arguments.